packages/playground/remote/src/lib/worker-thread.ts

```diff
@@ -256,6 +299,7 @@ try {
 	apiEndpoint.__internal_setRequestHandler(requestHandler);
 
 	const primaryPhp = await requestHandler.getPrimaryPhp();
+	downloadWordPressAssets(primaryPhp);
```

Comment:
How does this download apply for:
- browser storage with this download already applied
- device storage that already has full WP files
?

Reply:
TIL that device and browser storage should be roughly the same since Playground loads minified builds into device storage.

So the question should just be:
How does this download apply for browser and device storage with this download already applied?

Reply:
Yes, it will download the zip and add files to the FS. On future loads it will skip adding files if they already exists.

Reply:
> On future loads it will skip adding files if they already exists.

If you mean the `!php.fileExists(path))` check, it might still create some unwanted files if the user upgrades the WordPress version and the new one has a different file structure. We need something to keep track of these downloads to only do them once. How about removing entries from [/wordpress/wordpress-remote-asset-paths](https://github.com/WordPress/wordpress-playground/pull/1531) as those files get created?

Reply:
> Yes, it will download the zip and add files to the FS. On future loads it will skip adding files if they already exists.

This part kind of haunts me because it currently will download the zip each time Playground is loaded even if the files are already present.

> We need something to:
> - Avoid sending the fetch if we have all the static assets we need (e.g. we've already downloaded it in the browser storage mode or we're loading a full unminified WordPress build).

Maybe one possibility is iterating over the known remote asset paths set and checking to see what is left.

The set is added as part of #1539 which isn't merged yet:
https://github.com/WordPress/wordpress-playground/blob/ab5d83d20320a77f74eb809b1b496e4ebfaf8717/packages/playground/remote/src/lib/worker-thread.ts#L223

But if this PR can make it in before #1539, the `wordpress-remote-asset-paths` listing should exist after we retrieve it on demand. In that case, you you could also just read the file like the following and check static file existence based on that:
https://github.com/WordPress/wordpress-playground/blob/ab5d83d20320a77f74eb809b1b496e4ebfaf8717/packages/playground/remote/src/lib/worker-thread.ts#L308-L315

Reply:
Some thoughts for a later iteration:

At what point does it make sense to skip requesting the zip and just download the files themselves? What if we consume (and maybe save) the zip's central directory first which includes each files compressed size? At that point, we know:
- the size of the whole zip
- the path and compressed size of each file

Then we could consider make a judgement about whether to download the zip or just individual files. That may also make `wordpress-remote-asset-paths` redundant.

Reply:
@brandonpayton on a spectrum from "all or nothing" to "resume on error, download only what's needed", my preference is to do "all or nothing" because it's much simpler and the UX is mostly the same.

Reply:
[b2ba22b](https://github.com/WordPress/wordpress-playground/pull/1532/commits/b2ba22b64db60dc8d495cea3123771979e3258f1) will delete the `wordpress-remote-asset-paths` once all assets are downloaded. If `wordpress-remote-asset-paths` doesn't exist this means the files were downloaded and it won't run again.

Reply:
As mentioned in a [separate comment](https://github.com/WordPress/wordpress-playground/pull/1532#discussion_r1656396891), won't deleting the remote asset listing lead to it being auto-backfilled the next time a worker thread is initialized?

Reply:
You are correct, I switched to emptying it [de206a1](https://github.com/WordPress/wordpress-playground/pull/1532/commits/de206a14c7251a2fe246e080c8517318d5e242a4)